### PR TITLE
Add identity verifier interface

### PR DESCRIPTION
**Pull request by mlieberman85**
This will be useful if we verify identities through mechanisms
like GPG in addition to Sigstore

Signed-off-by: Michael Lieberman <mlieberman85@gmail.com>

### CODE REVIEW

File: pkg/ingestor/verifier/verifier.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// about the unverified identity.
+	Verify(payloadBytes []byte) ([]Identity, error)
+	Type() string
+}
+
+// Elements of this struct might be nil/empty if the key is invalid or the
+// Id of the identity can't be determined.
+type Identity struct {
+	Id       string
+	Key      crypto.PublicKey
```
Comment (lumjjb) [inline pkg/ingestor/verifier/verifier.go:40]: maybe add a comment here that this should only be used if `Verified` is true?

File: pkg/ingestor/verifier/verifier.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// verifiers that will check envelopes, signatures, and other payloads
+// for linking a payload back to an identity. In most cases this will be
+// done through verifying one or more signatures against known public keys.
+type Verifier interface {
+	// Verify takes in bytes and returns a list of identities. Those
+	// identities are marked as verified if the payload can be tied back to
+	// an identity, usually through signature validation. In cases where
+	// signatures can't be verified, it is still valuable to return data
+	// about the unverified identity.
+	Verify(payloadBytes []byte) ([]Identity, error)
```
Comment (lumjjb) [inline pkg/ingestor/verifier/verifier.go:32]: i think it looks good - i'm guessing all signature types would have a header to describe how to parse them right? or do we need to provide hints?

File: pkg/ingestor/verifier/verifier.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verifier
+
+import (
+	"crypto"
+)
+
+// The purpose of Verifier is to allow for multiple signature or identity
```
[the remark this answers is not in the document]
Comment (mlieberman85) [reply pkg/ingestor/verifier/verifier.go:22]: Yep, just learned about that lint. Will fix.

File: pkg/ingestor/verifier/verifier.go (hunk 1 above)
Comment (mlieberman85) [reply pkg/ingestor/verifier/verifier.go:40]: What do you mean?

File: pkg/ingestor/verifier/verifier.go (hunk 2 above)
Comment (mlieberman85) [reply pkg/ingestor/verifier/verifier.go:32]: Yeah, the way I see this getting used is, the ingestor would recognize that it's a DSSE, COSE, or another signature/envelope type and know it needs to verify it.

I also could see a future where there might be other ways of verifying identities as well that aren't purely through signature verification. e.g. verify that a specific key came from a specific root.

Comment (lumjjb) [reply pkg/ingestor/verifier/verifier.go:34]: ah ok - so would there be one Verifier for each type?

Comment (mlieberman85) [reply pkg/ingestor/verifier/verifier.go:34]: I would think so.

Comment (lumjjb) [reply pkg/ingestor/verifier/verifier.go:34]: ok yea thanks for clarifying! lgtm

**Review by lumjjb — APPROVED** [review]
lgtm

**Pull request merged**